chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -142,14 +161,46 @@ impl actix::Actor for PeerManagerActor {
             }
         }));
 
+        // Periodically update the connection store.
+        let clock = self.clock.clone();
+        let state = self.state.clone();
+        ctx.spawn(wrap_future(async move {
+            let mut interval = time::Interval::new(clock.now(), UPDATE_CONNECTION_STORE_INTERVAL);
+            loop {
+                state.update_connection_store(&clock);
+                interval.tick(&clock).await;
+            }
+        }));
+
+        // Periodically poll the connection store for connections we'd like to re-establish.
+        let clock = self.clock.clone();
+        let state = self.state.clone();
+        let addr = ctx.address();
+        ctx.spawn(wrap_future(async move {
+            let mut interval = time::Interval::new(clock.now(), POLL_CONNECTION_STORE_INTERVAL);
+            loop {
+                let pending_reconnect = state.poll_pending_reconnect();
+                for peer_info in pending_reconnect {
+                    if let Err(err) =
+                        addr.send(SpawnReconnectLoop { peer_info }.with_span_context()).await
```

Comment:
This thing is running within an arbiter, so you can spawn peer actors directly (and/or reconnect loop), rather than by sending a message to peer actor.